Auth docs update (#1807)
yaptide/ui: docs/authentication.md

    opt user doesn't have access
        AuthService ->> User: Message with access denied
    end
    loop Check backend availability every 3 minutes
        AuthService ->> Backend: (GET /)
        Backend ->> AuthService: Response with 'Hello World!'
    end
    loop Refresh backend token every backend access token lifetime
        AuthService ->> Backend: Refresh token (GET auth/keycloak)
        Backend ->> AuthService: Response with new backend access token in cookies and new accessExp
    end
    loop Refresh keycloak token every 1/3 of keycloak access tokens lifetime
        AuthService ->> Keycloak: Refresh token
        Keycloak ->> AuthService: Updated access token and refresh token
    end
    User ->> AuthService: Logout
    AuthService ->> Keycloak: Logout (GET /auth/realms/{our-realm}/protocol/openid-connect/logout)
    AuthService ->> Backend: Invalidate session (DELETE /auth/logout)
    Backend ->> AuthService: Response with cookies deleted
    AuthService ->> User: Clear user data
```

> -   **User requests login**: User initiates login with SSO.

Reviewer: You could use numbered list (best with manually hardcoded numbers), then those numbers could correspond to numbers of items on the diagram

            AuthService sets token refresh interval based on `accessExp` (backend token valid time).
            User receives authentication context.
        -   **If verification fails**: Backend sends "Forbidden" error due to invalid/expired token or Keycloak error.
    -   **If user lacks access**: AuthService informs user of access denial.

-   **Backend connection check**: Every 3 minutes, AuthService checks backend availability.
-   **Backend token periodic refresh**:

    -   Every backend token lifetime (`accessExp`), AuthService requests a token refresh from Backend.
    -   Backend responds with a new access token in cookies and an updated `accessExp`.

-   **Keycloak token periodic refresh**:

    -   Every 1/3 of the keycloak access token's lifetime, AuthService refreshes the Keycloak token.
    -   Keycloak provides an updated access token and refresh token to AuthService.

-   **Logout process**:
    -   User initiates logout.
    -   AuthService logs user out of Keycloak and clears user data.
    -   AuthService invalidates Backend session.
    -   Backend deletes session cookies.

## Our own database of users validation

> Overview of login and logout process while in demo or dev modes

Reviewer: BTW - do we have login process in demo mode ? I though that demo mode means login disables, like in github pages deployment,

Author: i meant dev deployment with docker. will fix

    end
    User ->> AuthService: Logout
    AuthService ->> Backend: Invalidate session (DELETE /auth/logout)
    Backend ->> AuthService: Response with cookies deleted
    AuthService ->> User: Clear User Data
```

-   **User login request**: User requests login through AuthService.

-   **Credential validation**:

    -   AuthService sends login credentials to Backend for validation.
    -   Backend responds with `accessExp`, setting access and refresh tokens in cookies.

-   **Provide authentication context**: AuthService provides the authenticated context to the user.

-   **Backend availability check**:

    -   Every 3 minutes, AuthService checks Backend availability.
    -   Backend responds with a "Hello World!" message.

-   **Backend token refresh**:

    -   Every 1/3 of `accessExp` (1/3 of Backend access token), AuthService refreshes the token with Backend.
    -   Backend returns a new access token in cookies and an updated `accessExp`.

-   **Logout process**:
    -   User initiates logout.
    -   AuthService sends a session invalidation request to Backend.
    -   Backend deletes session cookies, confirming logout.
    -   AuthService clears the user's data.

> ## Default vaules

Reviewer: ```suggestion
## Default values
```

---

### **Provide Authentication Context**

4️. **Provide auth context to user**

-   AuthService provides the authenticated context to the user.

---

### **Backend Availability Check**

5️. **Periodic availability check**

-   Every 3 minutes, AuthService checks Backend availability.

6️. **Backend response**

-   Backend responds with a "Hello World!" message to confirm availability.

---

### **Backend Token Refresh**

7️. **Periodic token refresh**

-   Every 1/3 of `accessExp` (1/3 of Backend access token lifetime), AuthService refreshes the token with Backend.

8️. **Receive new token and `accessExp`**

-   Backend returns a new access token in cookies and an updated `accessExp`.

---

### **Logout Process**

9️. **User initiates logout**

-   The user requests to log out.

1️0. **Invalidate session with Backend**

-   AuthService sends a session invalidation request to Backend.

1️1. **Session cookies deleted**

-   Backend deletes session cookies and confirms the logout.

1️2. **Clear user data**

-   AuthService clears the user's data from the local session.

---

## **Default Token Values**

> **Keycloak token values** (defined by Keycloak server on Ares):

Reviewer: ```suggestion
**Keycloak token values** (defined by Keycloak server in plgrid infrastructure `sso.plgrid.pl`):
```
